### PR TITLE
UsersTable: Render remove modal only once

### PR DESCRIPTION
**Pull request by dprokop**
Alternative to https://github.com/grafana/grafana/pull/43865

todo:
- [ ] Fix ServiceAccountsTable
- [ ] Add test for user removal

### CODE REVIEW

**Comment by eleijonmarck** [comment]
let's add service accounts in a additional PR which we can have a look at later.

**Comment by eleijonmarck** [comment]
@dprokop are these tests working? 🤔

File: public/app/features/users/UsersTable.tsx [hunk 1]
```diff
@@ -40,90 +39,103 @@ const UsersTable: FC<Props> = (props) => {
...
+                      disabled={!contextSrv.hasPermissionInMetadata(AccessControlAction.OrgUsersRoleUpdate, user)}
+                    />
+                  ) : (
+                    <OrgRolePicker
+                      aria-label="Role"
+                      value={user.role}
+                      disabled={!contextSrv.hasPermissionInMetadata(AccessControlAction.OrgUsersRoleUpdate, user)}
+                      onChange={(newRole) => onRoleChange(newRole, user)}
+                    />
+                  )}
```
Comment (eleijonmarck) [inline public/app/features/users/UsersTable.tsx:99]: not sure about this change?

File: public/app/features/users/UsersTable.tsx [hunk 2]
```diff
@@ -40,90 +39,103 @@ const UsersTable: FC<Props> = (props) => {
...
+          isOpen={true}
+          onConfirm={() => {
+            if (!userToRemove) {
+              return;
+            }
+            onRemoveUser(userToRemove);
+            setUserToRemove(null);
+          }}
+        />
+      )}
```
Comment (eleijonmarck) [inline public/app/features/users/UsersTable.tsx:137]: I like this addition! well done

File: public/app/features/users/UsersTable.tsx [hunk 3]
```diff
@@ -15,8 +15,7 @@ export interface Props {
 
 const UsersTable: FC<Props> = (props) => {
   const { users, orgId, onRoleChange, onRemoveUser } = props;
-
-  const [showRemoveModal, setShowRemoveModal] = useState(false);
+  const [userToRemove, setUserToRemove] = useState<OrgUser | null>(null);
```
Comment (eleijonmarck) [inline public/app/features/users/UsersTable.tsx:18]: this was a must! 🎉

**Review by kalleep — APPROVED** [review]
LGTM

**Review by eleijonmarck — APPROVED** [review]
🎉

File: public/app/features/users/UsersTable.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/users/UsersTable.tsx:98]: formatting only I think

**Pull request merged**